providers/Mora/mod.ts

```diff
@@ -0,0 +1,251 @@
+				}
+
+				throw new ResponseError(this.name, 'Failed to extract API arguments', webUrl);
+			},
+		}).then(({ content }) => {
+			const apiBase = apiUrl(content.mountPoint, content.labelId, content.materialNo);
+
+			const packageMetaUrl = new URL(apiBase);
+			packageMetaUrl.pathname += 'packageMeta.json';
+
+			return this.fetchJSON<WithApiUrl<Data>>(packageMetaUrl, {
+				policy: { maxTimestamp },
+				responseMutator: async (response) => {
+					const data = await response.json();
+					return new Response(
+						JSON.stringify({
+							apiUrl: apiBase,
+							data,
+						}),
+						response,
+					);
+				},
+			});
+		});
+	}
+}
+
+function apiUrl(mountPoint: string, labelId: string, materialNo: string) {
+	const paddedMaterialNo = materialNo.padStart(10, '0');
+	const slicedMaterialNo = `${paddedMaterialNo.slice(0, 4)}/${paddedMaterialNo.slice(4, 7)}/${
+		paddedMaterialNo.slice(7)
+	}`;
+
+	return new URL(`https://cf.mora.jp/contents/package/${mountPoint}/${labelId}/${slicedMaterialNo}/`);
+}
+
+export class MoraReleaseLookup extends ReleaseLookup<MoraProvider, PackageMeta> {
+	rawReleaseUrl: URL | undefined;
+	apiUrl: URL | undefined;
+
+	constructReleaseApiUrl(): URL | undefined {
+		return undefined;
+	}
+
+	async getRawRelease(): Promise<PackageMeta> {
+		if (this.lookup.method === 'gtin') {
+			throw new ProviderError(this.provider.name, 'GTIN lookups are not supported');
+		}
+
+		// Entity is already defined for ID/URL lookups.
+		const webUrl = this.provider.constructUrl(this.entity!);
+		this.rawReleaseUrl = webUrl;
+		const { content: release, timestamp } = await this.provider.extractEmbeddedJson<PackageMeta>(
+			webUrl,
+			this.options.snapshotMaxTimestamp,
+		);
+		this.apiUrl = release.apiUrl;
+		this.updateCacheTime(timestamp);
+
+		return release.data;
+	}
+
+	convertRawRelease(albumPage: PackageMeta): HarmonyRelease {
+		const label = { name: albumPage.labelcompanyname };
+		const tracklist = Object.entries(albumPage.trackList).map(([index, track]) =>
+			this.convertRawTrack(Number(index), track)
+		);
+		const types: ReleaseGroupType[] = [Object.keys(albumPage.trackList).length > 1 ? 'Album' : 'Single'];
+
+		// `distPartNo` *might* contain the GTIN, but will oftentimes contain either label-specific catalog numbers, or
+		// some package-specific code for mora (e.g. <package number>_F for FLAC releases)
+		let gtin = isValidGTIN(albumPage.distPartNo) ? albumPage.distPartNo : undefined;
+		if (!gtin) {
+			// If we're lucky, the GTIN might be in `cdPartNo`. In testing, the field seems to be `null` most of the time.
+			if (albumPage.cdPartNo && isValidGTIN(albumPage.cdPartNo)) {
+				gtin = albumPage.cdPartNo;
+			} else {
+				this.addMessage('Failed to determine GTIN', 'warning');
+			}
+		}
+
+		const release: HarmonyRelease = {
+			title: albumPage.title,
+			artists: [this.makeArtistCreditName(albumPage.artistName)],
+			labels: [label],
+			gtin,
+			releaseDate: parseISODateTime(albumPage.startDate),
```

Comment:
While this probably works as intended in UTC-N time zones, it returns the previous day in UTC+N time zones (since that function currently only expects the format from its descriptions).
I'm not sure whether we are supposed to interpret the 0:00 as UTC or as Japanese time... If it is UTC, the solution is as simple as the following hack (which we should document as accepted format):
```suggestion
			releaseDate: parseISODateTime(albumPage.startDate + ' +0'),
```

Reply:
Yeah, I've heard JS dates are weird, and since `parseISODateTime()` just delegates to the `Date` parser I figured it would *probably* support the date format on mora (`YYYY/MM/DD HH:mm:ss`).

No idea on the timezone, so I'll just use that hack for UTC.